pulpcore/pulpcore/tasking/tasks.py

```diff
@@ -200,20 +202,27 @@ def apply_async_with_reservation(self, resources, args=None, kwargs=None, **opti
         Returns (celery.result.AsyncResult):
             An AsyncResult instance as returned by Celery's apply_async
         """
-        resources = {util.get_url(resource) for resource in resources}
-        inner_task_id = str(uuid.uuid4())
+        resources = {util.get_url(resource) for resource in reservations}
         task_name = self.name
 
         # Set the parent attribute if being dispatched inside of a Task
         parent_arg = self._get_parent_arg()
 
-        TaskStatus.objects.create(pk=inner_task_id, state=TaskStatus.WAITING, **parent_arg)
+        if task_status is None:
```

Comment:
TODO
Task (task_status) should be created and passed in here everytime. Creation of CoreUpdateTask is temporary.